src/catchup/CatchupWork.cpp

```diff
@@ -281,6 +276,17 @@ CatchupWork::runCatchupStep()
                     mApp.getLedgerManager().getLastClosedLedgerHeader();
             }
         }
+        else if (mTransactionsVerifyApplySeq)
+        {
+            if (mTransactionsVerifyApplySeq->getState() ==
+                    State::WORK_SUCCESS &&
+                !mTransactionsVerifyEmitted)
+            {
+                mTransactionsVerifyEmitted = true;
```

Comment:
I don't think this codepath is reachable: if `mTransactionsVerifyApplySeq` is successful, then `mCatchupSeq` is also successful. `mCatchupSeq` status is checked above, so we can assert that `mTransactionsVerifyApplySeq` is successful and invoke the progress handler there (also, we don't need `mTransactionsVerifyEmitted` because `runCatchupStep` won't be scheduled anymore -- transaction application is the last step of catchup)

Reply:
`mApplyBufferedLedgersWork` is the last step now, which is why I invoke the progress handler before the buffered ledgers are applied. I verified this in the catchup test cases by throwing an exception in this block.